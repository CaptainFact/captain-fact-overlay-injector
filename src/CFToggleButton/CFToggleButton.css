.container {
  height: 30px;
  display: inline-block;
  background: #86d7ff;
  color: #2d2d2d;
  border-radius: 1em 0 0 1em;
  border: 1px solid lightgrey;
}

.container > * {
  max-height: 100%;
  vertical-align: middle;
}

.icon {
  height: 100%;
  margin-right: 5px;
}

.title {
  font-size: 16px;
  margin-right: 8px;
}

.radioBtn {
  height: 100%;
  max-height: 100%;
  min-width: 40px;
  text-align: center;
  display: inline-block;
  background: white;
  cursor: pointer;
  color: black;
  user-select: none;
}

.radioBtn:hover {
  color: grey;
}

.radioBtn.active {
  background-color: #3b84c4;
  box-shadow: inset 0 1px 10px hsla(0,0%,4%,.43);
  color: #fff;
}

.radioBtn span {
  padding: 8px;
  display: inline-block;
  font-size: 12px;
}