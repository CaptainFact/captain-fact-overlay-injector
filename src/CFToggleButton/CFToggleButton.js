import React from 'react'
import classNames from 'classnames'

import {container, icon, title, radioBtn, active} from './CFToggleButton.css'
import iconImg from '../assets/icon.png'


const LOCAL_STORAGE_KEY = 'captainfact_isActive'

export default class CFToggleButton extends React.PureComponent {
  constructor(props, context) {
    super(props, context)
    this.state = {isActive: loadIsActive()}
  }

  render() {
    return (
      <div className={container}>
        <img className={icon} src={iconImg}/>
        <span className={title}>CaptainFact</span>
        <div className={classNames(radioBtn, {[active]: this.state.isActive})} onClick={() => this.setActive(true)}>
          <span>ON</span>
        </div>
        <div className={classNames(radioBtn, {[active]: !this.state.isActive})} onClick={() => this.setActive(false)}>
          <span>OFF</span>
        </div>
      </div>
    )
  }

  setActive(isActive) {
    this.setState({isActive})
    saveIsActive(isActive)
  }
}

const loadIsActive = () =>
  localStorage.getItem(LOCAL_STORAGE_KEY) !== 'false'

const saveIsActive = (isActive) =>
  localStorage.setItem(LOCAL_STORAGE_KEY, isActive ? 'true' : 'false')