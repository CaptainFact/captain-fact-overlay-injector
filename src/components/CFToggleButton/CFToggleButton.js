import React from 'react'
import { connect } from 'react-redux'
import classNames from 'classnames'

import {container, icon, title, radioBtn, active} from './CFToggleButton.css'
import iconImg from '../../assets/icon.png'


export class CFToggleButton extends React.PureComponent {
  render() {
    return (
      <div className={container}>
        <img className={icon} src={iconImg}/>
        <a href="https://captainfact.io" className={title}>CaptainFact</a>
        <div className={classNames(radioBtn, {[active]: this.props.isEnabled})} onClick={() => this.setEnabled(true)}>
          <span>ON</span>
        </div>
        <div className={classNames(radioBtn, {[active]: !this.props.isEnabled})} onClick={() => this.setEnabled(false)}>
          <span>OFF</span>
        </div>
      </div>
    )
  }

  setEnabled(enabled) {
    if (!this.props.isEnabled && enabled)
      this.props.enable()
    else if (this.props.isEnabled && !enabled)
      this.props.disable()
  }
}

export default connect(state => ({isEnabled: state.Interface.isEnabled}))(CFToggleButton)