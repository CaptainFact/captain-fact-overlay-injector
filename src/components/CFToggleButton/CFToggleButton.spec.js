import {CFToggleButton} from './CFToggleButton'


test('button activated', () => {
  snapshot(<CFToggleButton isEnabled={true}/>)
})

test('button disabled', () => {
  snapshot(<CFToggleButton isEnabled={false}/>)
})

test('displays custom icon', () => {
  snapshot(<CFToggleButton isEnabled={true} icon="custom-icon.jpg"/>)
})

test('verify buttons labels and order', () => {
  const mounted = mount(<CFToggleButton isEnabled={false}/>)
  expect(mounted.find('.radioBtn').at(0).text()).toBe('ON')
  expect(mounted.find('.radioBtn').at(1).text()).toBe('OFF')
})

test('call enable if disabled', () => {
  const enableFunc = jest.fn()
  const mounted = mount(<CFToggleButton isEnabled={false} enable={enableFunc}/>)

  mounted.find('.radioBtn').at(0).simulate('click')
  expect(enableFunc).toBeCalled()
})

test('call enable if enabled', () => {
  const disableFunc = jest.fn()
  const mounted = mount(<CFToggleButton isEnabled={false} enable={disableFunc}/>)

  mounted.find('.radioBtn').at(0).simulate('click')
  expect(disableFunc).toBeCalled()
})
