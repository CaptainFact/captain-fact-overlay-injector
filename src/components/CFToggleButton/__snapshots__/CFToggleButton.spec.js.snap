// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button activated 1`] = `
<div
  className="container"
>
  <img
    className="icon"
    src="test-file-stub"
  />
  <a
    className="title"
    href="https://captainfact.io"
  >
    CaptainFact
  </a>
  <div
    className="radioBtn active"
    onClick={[Function]}
  >
    <span>
      ON
    </span>
  </div>
  <div
    className="radioBtn"
    onClick={[Function]}
  >
    <span>
      OFF
    </span>
  </div>
</div>
`;

exports[`button disabled 1`] = `
<div
  className="container"
>
  <img
    className="icon"
    src="test-file-stub"
  />
  <a
    className="title"
    href="https://captainfact.io"
  >
    CaptainFact
  </a>
  <div
    className="radioBtn"
    onClick={[Function]}
  >
    <span>
      ON
    </span>
  </div>
  <div
    className="radioBtn active"
    onClick={[Function]}
  >
    <span>
      OFF
    </span>
  </div>
</div>
`;
